core/templates/dev/head/pages/collection_editor/collection_editor.html

```diff
@@ -117,6 +117,9 @@
 
   {# TODO(bhenning): Remove these once RouterServices is decoupled from the exploration editor #}
   <script src="{{TEMPLATE_DIR_PREFIX}}/pages/exploration_editor/EditorServices.js"></script>
+  <script src="{{TEMPLATE_DIR_PREFIX}}/pages/exploration_editor/ExplorationDataService.js"></script>
+  <script src="{{TEMPLATE_DIR_PREFIX}}/pages/exploration_editor/EditorStateService.js"></script>
+  <script src="{{TEMPLATE_DIR_PREFIX}}/pages/exploration_editor/AngularNameService.js"></script>
```

Comment:
Are these really needed here ?

Reply:
Done. They are removed now. 

If you are checking if these services are dependent on another, should I generally just do a grep on the file names and see if they are referenced by files within another directory i.e. collection_editor?

Reply:
Oh, I meant you had to analyse it before removing! I don't know for sure whether all would be removed or not. 
Check out [this](https://github.com/oppia/oppia/pull/3938/files/972285acb5956c1423e89c33592f565077ff7eb7), here I have tried to explain how I would go about analysing it. (See the comments for file `admin.html`). Let me know if you have any questions :)

Reply:
Thanks for the link with the great explanation. 
So ExplorationDataService is a dependency within EditorServices; EditorStateService is a dependency within RouterService; and AngularNameService is a dependency within EditorServices. EditorServices and RouterService are imported in collection_editor, so I believe they are needed here if my thinking is correct?

Reply:
Hi @bching, yup as much as I've checked these are needed here. 
In general, you had to see, for example, if we need to include AngularNameService.js:
-AngularNameService is being directly used anywhere in collection_editor (say, there was a direct mention of AngularNameService in any of the services defined in collection_editor)
or
-Any other service which depends on AngularNameService (directly or indirectly) is being used in collection_editor.

We have AngularNameService defined in EditorServices.js and it is a dependency in ExplorationStateService (also defined in EditorServices.js). Now it might not be enough to say that as EditorServices.js is imported in collection_editor, so will also need AngularNameService.js. As it can be the case,(just for example sake consider), that in collection_editor we were using the `changeListService` only, because of which we needed to import EditorServices but then there would have been no need of importing AngularNameService too as changeListService does not depend on it.  
In short you had to see the subset of services defined in EditorServices.js that depend on AngularNameService and whether those services were being used in collection_editor.

The reason why we need AngularNameService is because `explorationStatesService` has it as a dependency, and RouterService (defined in RouterService.js) has `explorationStatesService` as a dependency (so indirect dependency on AngularNameService). And in collection_editor we need RouterService.js, so we check which service defined in RouterService.js is actually needed by collection_editor. But as there is just one service present in RouterService.js we know for sure that `RouterService` is the one needed by collection_editor. So we can now say for sure that AngularNameService.js is also required by collection_editor.

Hope I haven't confused you :)

Reply:
Then, one possible outcome of extracting these services into discrete files is a reduction in the number of imports in a page? 

That was a good explanation. Thank you for clearing things up further.

Reply:
Well yup, reduction in some pages and increase for some. I think the advantage of having separate file for each service, in terms of import would be that now each page would be including only the services really needed by it.